doc/glossary.rst

```diff
@@ -205,6 +205,23 @@ General Concepts
         exceptional behaviours on the estimator using semantic :term:`estimator
         tags`.
 
+    cross-fitting
+    cross fitting
+        A resampling method that iteratively partitions data into complementary
+        'train' and 'estimation' subsets. The 'estimation' subset may be used
+        to estimate a parameter or predict a target. The outputs from
+        each iteration are combined and used in a downstream prcess, generally
+        to avoid bias.
```

Comment:
Maybe it would be useful to cross-link to typical (meta-)estimators that implement this strategy (e.g. StackingRegressor/Classifier and TargetEncoder, CalibratedClassifierCV...).

Note: the current implementation of `CalibratedClassifierCV` is a bit different from the others because it does not attempt to concatenate the predictions of the first stage to fit the second stage (but instead fits one second stage estimator per cross-fitting iteration and then averages the predictions of the second stage model). Still, I think those are morally all examples of a cross-fitting strategy.

Another example not yet implemented in scikit-learn are honest trees and forests where the values of the leafs are estimated on different training samples that those used to learn the decision splits of the trees.

Reply:
Good idea, I meant to do this but  forgot!